Sources/SwiftyXPC/XPCFileDescriptor.swift

```diff
@@ -19,4 +26,12 @@ public final class XPCFileDescriptor: Codable {
     public init(fileDescriptor: Int32) {
         self.fileDescriptor = fileDescriptor
     }
+    
+    public func dup() -> Int32 {
+        Darwin.dup(fileDescriptor)
+    }
```

Comment:
If we are going to be targeting 11.0 and above with this class, we should support `FileDescriptor` as well. Also, `dup()` can return an error; we shouldn't ignore that.

```suggestion
    @available(macOS 11.0, *)
    public init(fileDescriptor: FileDescriptor) {
        self.fileDescriptor = fileDescriptor.rawValue
    }
    
    public func dup() throws -> Int32 {
        let fd = Darwin.dup(fileDescriptor)

        if fd < 0 {
            if #available(macOS 11.0, *) {
                throw Errno(rawValue: errno)
            } else {
                throw XPCErrorRegistry.BoxedError(domain: "NSPOSIXErrorDomain", code: Int(errno))
            }
        }

        return fd
    }

    @available(macOS 11.0, *)
    public func duplicate() throws -> FileDescriptor {
        if #available(macOS 12.0, *) {
            return try FileDescriptor(rawValue: self.fileDescriptor).duplicate()
        } else {
            return try FileDescriptor(rawValue: self.dup())
        }
    }
```